timetracker/entries/forms.py

```diff
@@ -39,11 +51,15 @@ def clean(self):
         # is preserved
         cleaned_data = super(EntryForm, self).clean()
 
-        # Get the start and end values from the cleaned_data dictionary, or None
+        # Get the start and stop values from the cleaned_data dictionary, or None
         # if the dictionary keys are missing
         start = cleaned_data.get('start', None)
-        end = cleaned_data.get('end', None)
+        stop = cleaned_data.get('stop', None)
+        if 'create_endnow' in self.data:
+            stop = timezone.now()
+            cleaned_data['stop'] = stop
```

Comment:
Generally it's best practice not to change any data in the a form's clean() method. It would be better here to define a clean_stop() method, similar to clean_start that sets the timezone.now() value.